src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1183,26 +1183,49 @@ CanUseExclusiveConnections(Oid relationId, bool localTableEmpty)
 	}
 	else if (shouldRunSequential && ParallelQueryExecutedInTransaction())
 	{
-		char *relationName = get_rel_name(relationId);
-
 		/*
-		 * If there has already been a parallel query executed, the sequential mode
-		 * would still use the already opened parallel connections to the workers,
-		 * thus contradicting our purpose of using sequential mode.
+		 * We decided to use sequential execution. It's either because relation
```

Comment:
why do we  touch this part of the code  in this PR?  Even if there are  problems, we'd better resolve  in different PRs as these code paths are  tricky to  change, may have unexpected results.


We should find a test that this PR  (only the  local execution part) improves.

assume  a single node cluster, on master this works fine, but should have been 
```SQL
-- find a test  which already does this
SELECT 1 FROM master_add_node('localhost', 5432, groupid => 0);

CREATE TABLE distributed_table(id int PRIMARY KEY, value_1 int);
SELECT create_distributed_table('distributed_table', 'id');

CREATE TABLE reference_table(id int PRIMARY KEY);
SELECT create_reference_table('reference_table');
ALTER TABLE distributed_table ADD CONSTRAINT c_check FOREIGN KEY (value_1) REFERENCES reference_table(id);


postgres=# set client_min_messages TO DEBUG1;
SET
postgres=# SET citus.log_remote_commands TO ON;
SET
SELECT count(*) FROM reference_table;
NOTICE:  executing the command locally: SELECT count(*) AS count FROM public.reference_table_102106 reference_table
 count 
-------
     0
(1 row)

-- does a parallel query
ALTER TABLE distributed_table DROP COLUMN value_1;


-- on the master branch, this shows up as
-- parallel, on your branch it is sequential
-- and this is  what  we wantt  to achieve
 show   citus.multi_shard_modify_mode ;
 citus.multi_shard_modify_mode 
-------------------------------
 parallel
(1 row)
```

And,  on master this works, butt  on your branch it should  fail

```SQL
BEGIN;
ALTER TABLE distributed_table ADD COLUMN value_2 int;
SELECT count(*) FROM reference_table;
ERROR:  cannot execute SELECT on reference table "reference_table" because there was a parallel DDL access to distributed table "distributed_table" in the same transaction
HINT:  Try re-running the transaction with "SET LOCAL citus.multi_shard_modify_mode TO 'sequential';"
```